File: src/main/java/me/elsiff/morefish/hooker/CitizensHooker.java
package me.elsiff.morefish.hooker;

import net.citizensnpcs.api.CitizensAPI;
import net.citizensnpcs.api.trait.TraitInfo;

public class CitizensHooker {
    private TraitInfo fishShop;

    public CitizensHooker() {
        this.fishShop = TraitInfo.create(FishShopTrait.class).withName("fishshop");
    }

    public void registerTrait() {
        CitizensAPI.getTraitFactory().registerTrait(fishShop);
    }

    public void deregisterTrait() {
        CitizensAPI.getTraitFactory().deregisterTrait(fishShop);
    }
}

File: src/main/kotlin/me/elsiff/morefish/hooker/CitizensHooker.kt
package me.elsiff.morefish.hooker

import me.elsiff.morefish.MoreFish
import me.elsiff.morefish.shop.FishShopKeeperTrait
import net.citizensnpcs.api.CitizensAPI
import net.citizensnpcs.api.trait.TraitInfo

/**
 * Created by elsiff on 2019-01-24.
 */
class CitizensHooker : PluginHooker {
    override val pluginName = "Citizens"
    override var hasHooked = false
    private lateinit var traitInfo: TraitInfo

    override fun hook(plugin: MoreFish) {
        traitInfo = TraitInfo.create(FishShopKeeperTrait::class.java)
        CitizensAPI.getTraitFactory().registerTrait(traitInfo)
        FishShopKeeperTrait.init(plugin.fishShop)
        hasHooked = true
    }

    fun dispose() {
        CitizensAPI.getTraitFactory().deregisterTrait(traitInfo)
    }
}
